pkgs/applications/networking/taler/default.nix

```diff
@@ -1,57 +1,104 @@
 { lib, stdenv, fetchurl, curl, gnunet, jansson, libgcrypt, libmicrohttpd
-, qrencode, libsodium, libtool, pkg-config, postgresql, sqlite }:
+, qrencode, libsodium, libtool, pkg-config, postgresql, sqlite, autoreconfHook, jinja2, recutils, wget, jq, gettext, texinfo
+}:
 
 let
   gnunet' = gnunet.override { postgresqlSupport = true; };
-
-  mkTaler = { pname, version, sha256 }:
-    extraAttrs:
-    stdenv.mkDerivation (extraAttrs // {
-      inherit pname version;
-      src = fetchurl {
-        url = "mirror://gnu/taler/${pname}-${version}.tar.gz";
-        inherit sha256;
-      };
-      enableParallelBuilding = true;
-      meta = with lib; {
-        description = "Anonymous, taxable payment system.";
-        homepage = "https://taler.net/";
-        license = licenses.agpl3Plus;
-        maintainers = with maintainers; [ ehmry ];
-        platforms = platforms.gnu ++ platforms.linux;
-      };
-    });
-
 in rec {
-
-  taler-exchange = mkTaler {
+  taler-exchange = stdenv.mkDerivation rec {
     pname = "taler-exchange";
-    version = "0.8.1";
-    sha256 = "sha256-MPt3n1JXd0Y89b1qCuF6YxptSr7henfYp97JTq1Z+x4=";
-  } {
+    version = "0.8.5";
+    src = fetchurl {
+      url = "mirror://gnu/taler/${pname}-${version}.tar.gz";
+      sha256 = "sha256-jehQKkDYdDkEXen/LSn0gZi4tYpykYM92W35ZbSLNj4=";
+    };
+    postPatch = ''
+      find . -name '*.sh' -exec sed -i 's%#!/bin/bash%#!${stdenv.shell}%g' {} \;
+    '';
+    enableParallelBuilding = true;
+    nativeBuildInputs = [
+      autoreconfHook
+    ];
     buildInputs = [
-      curl
-      jansson
       libgcrypt
       libmicrohttpd
+      jansson
       libsodium
-      libtool
       postgresql
-      # sqlite
+      curl
+      jinja2
+      recutils
+      gettext
+      texinfo # Fix 'makeinfo' is missing on your system.
     ];
     propagatedBuildInputs = [ gnunet' ];
-    patches = [ ./exchange-fix-6665.patch ];
+    configureFlags = [ "--with-gnunet=${gnunet'}" ];
+    checkInputs = [ wget curl ];
+    meta = with lib; {
```

Comment:
meta should come after the installCheck.

Reply:
done